packages/aws-cdk-lib/core/lib/stack.ts

```diff
@@ -1046,7 +1073,8 @@ export class Stack extends Construct implements ITaggable {
         Annotations.of(this).addInfo(`Number of resources: ${numberOfResources} is approaching allowed maximum of ${this.maxResources}`);
       }
     }
-    fs.writeFileSync(outPath, JSON.stringify(template, undefined, 1));
+    const indent = this._suppressTemplateIndentation ? undefined : 1;
+    fs.writeFileSync(outPath, JSON.stringify(template, undefined, indent));
```

Comment:
Can we check the size of the template and if it is too large throw a warning that you should set the `suppressTemplateIndentation` flag or split the resources between stacks.